src/webui/abstractwebapplication.cpp

```diff
@@ -234,14 +234,15 @@ QString AbstractWebApplication::generateSid()
 
 void AbstractWebApplication::translateDocument(QString& data)
 {
-    const QRegExp regex("QBT_TR\\((([^\\)]|\\)(?!QBT_TR))+)\\)QBT_TR");
+    const QRegExp regex("QBT_TR\\((([^\\)]|\\)(?!QBT_TR))+)\\)QBT_TR(\\[CONTEXT=([a-zA-Z_][a-zA-Z0-9_]*)([\\]]))?");
```

Comment:
IMO, it's better than your original proposal. I would like to know other opinions. Can you present discussed formats (with your and my suggestions) in PR description and call other developers too?